src/api/users/users.services.ts

```diff
@@ -0,0 +1,79 @@
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+
+// gets all the users from the db
+export const getAllUsers = () => {
+  return prisma.users.findMany({
+    select: {
+      name: true,
+      last_name: true,
+      city: true,
+      user_role: true
+    }
+  });
+}
+
+
+// get a single user by the id 
+
+export const getUserById = (id: string) => {
+  return prisma.users.findUnique({
+    where: {
+      user_id:id
+    }
+  });
+}
+
+export const createUser = (input: any) => {
+  const dateOfBirth = new Date(input.date_of_birth);
+  return prisma.users.create({
+    data: {
+      email: input.email,
+      password: input.password,
+      name: input.name,
+      last_name: input.last_name,
+      document_type: input.document_type,
+      document_number: input.document_number,
+      date_of_birth: dateOfBirth,
+      city: input.city,
+      contact_email: Boolean(input.contact_email),
+      contact_sms: Boolean(input.contact_sms),
+      contact_wpp: Boolean(input.contact_wpp),
+      user_role: input.user_role
+    }
```

Comment:
The input could be destructured and the could could be the following:
```js
const dateOfBirth = new Date(date_of_birth);
 return prisma.users.create({
    data: {
      email,
      password,
      name,
      last_name,
      document_type,
      document_number,
      date_of_birth: dateOfBirth,
      city
      contact_email: Boolean(contact_email),
      contact_sms: Boolean(contact_sms),
      contact_wpp: Boolean(contact_wpp),
      user_role
    }
```

Reply:
resolved